backend/announcements/serializers.py

```diff
@@ -0,0 +1,57 @@
+from announcements.models import Announcement, Audience
+from rest_framework import serializers
+
+
+class AudienceSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Audience
+        fields = ("name",)
+
+
+class AnnouncementSerializer(serializers.ModelSerializer):
+    audiences = serializers.SlugRelatedField(
+        many=True, slug_field="name", queryset=Audience.objects.all()
+    )
+
+    class Meta:
+        model = Announcement
+        fields = (
+            "id",
+            "title",
+            "message",
+            "announcement_type",
+            "release_time",
+            "end_time",
+            "audiences",
+        )
+
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation["audiences"] = [
+            audience.name for audience in instance.audiences.all()
+        ]
+        return representation
+
+    def to_internal_value(self, data):
+        audiences = data.get("audiences")
+        if isinstance(audiences, list):
+            audience_objs = []
+            for audience_name in audiences:
+                audience = Audience.objects.filter(name=audience_name).first()
+                if audience:
```

Comment:
Not entirely sure if we want to silently let this error pass, might need to raise a serializer error. What do you think?